Exporter crashes on region quota query

# Bug Report

Upon failure when querying region quotas the exporter crashes.

## Expected Behaviour

It should retry using an exponential back-off.

## Actual Behaviour

It crashes completely with `invalid memory address or nil pointer dereference`:

```
time="2020-06-26T12:47:11Z" level=info msg="Starting gcp_quota_exporter (version=, branch=, revision=)" source="main.go:164"
time="2020-06-26T12:47:11Z" level=info msg="Build context (go=go1.13.12, user=, date=)" source="main.go:165"
time="2020-06-26T12:47:16Z" level=info msg="Google Project: XXXXXX" source="main.go:210"
time="2020-06-26T12:47:16Z" level=info msg="Listening on :9592" source="main.go:211"
time="2020-06-26T12:51:09Z" level=error msg="Failure when querying region quotas: Get https://www.googleapis.com/compute/v1/projects/XXXXXX/regions?alt=json&prettyPrint=false: net/http: request canceled (Client.Timeout exceeded while awaiting headers)" source="main.go:73"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x88 pc=0xb1a206]

goroutine 124 [running]:
main.(*Exporter).Collect(0xc0002da090, 0xc00036e120)
	/app/main.go:94 +0xc6
github.com/prometheus/client_golang/prometheus.(*Registry).Gather.func1()
	/go/pkg/mod/github.com/prometheus/client_golang@v0.9.3/prometheus/registry.go:434 +0x19d
created by github.com/prometheus/client_golang/prometheus.(*Registry).Gather
	/go/pkg/mod/github.com/prometheus/client_golang@v0.9.3/prometheus/registry.go:526 +0xe12
```

--- a/main.go
+++ b/main.go
@@ -71,7 +71,7 @@ func (e *Exporter) scrape() (up float64, prj *compute.Project, rgl *compute.Regi
 	regionList, err := e.service.Regions.List(e.project).Do()
 	if err != nil {
 		log.Errorf("Failure when querying region quotas: %v", err)
-		return 0, nil, nil
+		return 0, project, nil
 	}
 
 	return 1, project, regionList
@@ -91,16 +91,20 @@ func (e *Exporter) Collect(ch chan<- prometheus.Metric) {
 
 	up, project, regionList := e.scrape()
 
-	for _, quota := range project.Quotas {
-		ch <- prometheus.MustNewConstMetric(limitDesc, prometheus.GaugeValue, quota.Limit, e.project, "", quota.Metric)
-		ch <- prometheus.MustNewConstMetric(usageDesc, prometheus.GaugeValue, quota.Usage, e.project, "", quota.Metric)
+	if project != nil {
+		for _, quota := range project.Quotas {
+			ch <- prometheus.MustNewConstMetric(limitDesc, prometheus.GaugeValue, quota.Limit, e.project, "", quota.Metric)
+			ch <- prometheus.MustNewConstMetric(usageDesc, prometheus.GaugeValue, quota.Usage, e.project, "", quota.Metric)
+		}
 	}
 
-	for _, region := range regionList.Items {
-		regionName := region.Name
-		for _, quota := range region.Quotas {
-			ch <- prometheus.MustNewConstMetric(limitDesc, prometheus.GaugeValue, quota.Limit, e.project, regionName, quota.Metric)
-			ch <- prometheus.MustNewConstMetric(usageDesc, prometheus.GaugeValue, quota.Usage, e.project, regionName, quota.Metric)
+	if regionList != nil {
+		for _, region := range regionList.Items {
+			regionName := region.Name
+			for _, quota := range region.Quotas {
+				ch <- prometheus.MustNewConstMetric(limitDesc, prometheus.GaugeValue, quota.Limit, e.project, regionName, quota.Metric)
+				ch <- prometheus.MustNewConstMetric(usageDesc, prometheus.GaugeValue, quota.Usage, e.project, regionName, quota.Metric)
+			}
 		}
 	}
 

--- a/main_test.go
+++ b/main_test.go
@@ -8,7 +8,7 @@ import (
 func TestScrape(t *testing.T) {
 
 	// TestSuccessfulConnection
-	exporter, _ := NewExporter("credentials.json", os.Getenv("GCP_PROJECT"))
+	exporter, _ := NewExporter(os.Getenv("GCP_PROJECT"))
 	up, _, _ := exporter.scrape()
 	if up == 0 {
 		t.Errorf("TestSuccessfulConnection: up=%v, expected=1", up)
@@ -16,7 +16,7 @@ func TestScrape(t *testing.T) {
 
 	// TestFailedConnection
 	// Set the project name to "503" since the Google Compute API will append this to the end of the BasePath
-	exporter, _ = NewExporter("credentials.json", "503")
+	exporter, _ = NewExporter("503")
 	exporter.service.BasePath = "https://httpbin.org/status/"
 	up, _, _ = exporter.scrape()
 	if up != 0 {

